ValueError: 'mech_price' field is not set for this period [5791] and no default value was provided

```
File "/root/agent/vendor/valory/connections/p2p_libp2p_client/connection.py", line 268, in close
    await self.pipe.close()
  File "/usr/local/lib/python3.11/site-packages/aea/helpers/pipe.py", line 600, in close
    await self._sock.close()
  File "/usr/local/lib/python3.11/site-packages/aea/helpers/pipe.py", line 327, in close
  File "/usr/local/lib/python3.11/asyncio/streams.py", line 361, in wait_closed
    await self._protocol._get_close_waiter(self)
  File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 970, in _read_ready__get_buffer
    nbytes = self._sock.recv_into(buf)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
ConnectionResetError: [Errno 104] Connection reset by peer
    await self._writer.wait_closed()
  File "/usr/local/lib/python3.11/asyncio/streams.py", line 361, in wait_closed
    await self._protocol._get_close_waiter(self)
  File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 970, in _read_ready__get_buffer
    nbytes = self._sock.recv_into(buf)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
ConnectionResetError: [Errno 104] Connection reset by peer
[2024-01-09 00:04:31,384] [ERROR] [agent] Error while disconnecting <class 'connection_module.P2PLibp2pClientConnection'>: [Errno 104] Connection reset by peer
Traceback (most recent call last):
[2024-01-09 00:04:31,384][ERROR] [agent] Error while disconnecting <class 'connection_module.P2PLibp2pClientConnection'>: [Errno 104] Connection reset by peer
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aea/multiplexer.py", line 454, in _disconnect_all
    await asyncio.wait_for(
  File "/usr/local/lib/python3.11/asyncio/tasks.py", line 489, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aea/multiplexer.py", line 481, in _disconnect_one
    await connection.disconnect()
  File "/usr/local/lib/python3.11/site-packages/aea/multiplexer.py", line 454, in _disconnect_all
    await asyncio.wait_for(
  File "/usr/local/lib/python3.11/asyncio/tasks.py", line 489, in wait_for
    return fut.result()
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aea/multiplexer.py", line 481, in _disconnect_one
    await connection.disconnect()
  File "/root/agent/vendor/valory/connections/p2p_libp2p_client/connection.py", line 501, in disconnect
    await self._node_client.close()
  File "/root/agent/vendor/valory/connections/p2p_libp2p_client/connection.py", line 268, in close
  File "/root/agent/vendor/valory/connections/p2p_libp2p_client/connection.py", line 501, in disconnect
    await self._node_client.close()
  File "/root/agent/vendor/valory/connections/p2p_libp2p_client/connection.py", line 268, in close
    await self.pipe.close()
  File "/usr/local/lib/python3.11/site-packages/aea/helpers/pipe.py", line 600, in close
    await self._sock.close()
  File "/usr/local/lib/python3.11/site-packages/aea/helpers/pipe.py", line 327, in close
    await self._writer.wait_closed()
  File "/usr/local/lib/python3.11/asyncio/streams.py", line 361, in wait_closed
    await self._protocol._get_close_waiter(self)
  File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 970, in _read_ready__get_buffer
    nbytes = self._sock.recv_into(buf)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
ConnectionResetError: [Errno 104] Connection reset by peer
    await self.pipe.close()
  File "/usr/local/lib/python3.11/site-packages/aea/helpers/pipe.py", line 600, in close
    await self._sock.close()
  File "/usr/local/lib/python3.11/site-packages/aea/helpers/pipe.py", line 327, in close
    await self._writer.wait_closed()
  File "/usr/local/lib/python3.11/asyncio/streams.py", line 361, in wait_closed
    await self._protocol._get_close_waiter(self)
  File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 970, in _read_ready__get_buffer
    nbytes = self._sock.recv_into(buf)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
ConnectionResetError: [Errno 104] Connection reset by peer
Stopping AEA 'agent' ...
AEA 'agent' stopped.
Error: AEA was terminated cause exception `An error occurred during act of behaviour valory/trader_abci:0.1.0/TraderConsensusBehaviour:
Traceback (most recent call last):

  File "/root/agent/vendor/valory/skills/abstract_round_abci/base.py", line 594, in get
    raise ValueError(

ValueError: 'mech_price' field is not set for this period [5791] and no default value was provided.


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/usr/local/lib/python3.11/site-packages/aea/skills/base.py", line 440, in act_wrapper
    raise AEAActException(

aea.exceptions.AEAActException: An error occurred during act of behaviour valory/trader_abci:0.1.0/RedeemBehaviour:
Traceback (most recent call last):

  File "/root/agent/vendor/valory/skills/abstract_round_abci/base.py", line 594, in get
    raise ValueError(

ValueError: 'mech_price' field is not set for this period [5791] and no default value was provided.

` in skills <bound method Behaviour.act_wrapper of <behaviours.TraderConsensusBehaviour object at 0x7fc88eaba810>>! Please check logs.
[2024-01-09 00:04:32,098][ERROR] Task was destroyed but it is pending!
task: <Task pending name='Task-3722233' coro=<TcpServerChannel.receive_messages() running at /root/agent/vendor/valory/connections/abci/connection.py:954> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[StreamReaderProtocol.connection_made.<locals>.callback() at /usr/local/lib/python3.11/asyncio/streams.py:248]>
[2024-01-09 00:04:32,098][ERROR] Task was destroyed but it is pending!
task: <Task pending name='Task-3722235' coro=<TcpServerChannel.receive_messages() running at /root/agent/vendor/valory/connections/abci/connection.py:954> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[StreamReaderProtocol.connection_made.<locals>.callback() at /usr/local/lib/python3.11/asyncio/streams.py:248]>
[2024-01-09 00:04:32,098][ERROR] Task was destroyed but it is pending!
task: <Task pending name='Task-3722237' coro=<TcpServerChannel.receive_messages() running at /root/agent/vendor/valory/connections/abci/connection.py:954> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[StreamReaderProtocol.connection_made.<locals>.callback() at /usr/local/lib/python3.11/asyncio/streams.py:248]>
[2024-01-09 00:04:32,098][ERROR] Task was destroyed but it is pending!
task: <Task pending name='Task-3722239' coro=<TcpServerChannel.receive_messages() running at /root/agent/vendor/valory/connections/abci/connection.py:954> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[StreamReaderProtocol.connection_made.<locals>.callback() at /usr/local/lib/python3.11/asyncio/streams.py:248]>
```

--- a/packages/packages.json
+++ b/packages/packages.json
@@ -1,13 +1,13 @@
 {
     "dev": {
         "skill/valory/market_manager_abci/0.1.0": "bafybeifdf6uoscrgvlgqceqcph2pgoygobznw6hbodpav7lubbqsyh4kby",
-        "skill/valory/decision_maker_abci/0.1.0": "bafybeiaan3hccyiocw6a2hecrcfx47sjxkuerbcoxjwjgr32zuqujs3qsu",
-        "skill/valory/trader_abci/0.1.0": "bafybeieb3pleydyefkxd7hcern2ghrlxcyasw2hq76l7lucogtgfjimchy",
+        "skill/valory/decision_maker_abci/0.1.0": "bafybeie6o4n3s557caqxcqtdxquiej6yakwc3z46anbnfnkaguvxkqjkhq",
+        "skill/valory/trader_abci/0.1.0": "bafybeigrm3n43ycgwysx3ovh67drnheyt5c2gqykb7bp7g6eqviyy3utom",
         "contract/valory/market_maker/0.1.0": "bafybeiain372i5t6eteem36vzhwh6ppka6hc4oyrxjtyzuar6rd6bjocem",
-        "agent/valory/trader/0.1.0": "bafybeieib2wgae3hgxshw2n6kmii5m76awafe6f46mrkif7wtc5mncvdlm",
-        "service/valory/trader/0.1.0": "bafybeicommzb25fe2df6w5x4w5jqxjybgth3nagssvdn4xf2vqq6gf5t2e",
+        "agent/valory/trader/0.1.0": "bafybeidjit4jcfqr56mcy65jla76g7jiz7nybfoeycftfijtutfqcorfsa",
+        "service/valory/trader/0.1.0": "bafybeibvduizyhrd7g37wfnvxfw4r3tbgrpjtkpyl7jletubtn3flmhj2i",
         "contract/valory/erc20/0.1.0": "bafybeiad6qbyhr3sgvclqne7tvyvqt32pcj5aq47hhkd2vmawaflhy5vna",
-        "skill/valory/tx_settlement_multiplexer_abci/0.1.0": "bafybeigvuuvfy7dcdp2kofaorjrqhfxiqdcjsojb4xlcim5x35dmmtzyae",
+        "skill/valory/tx_settlement_multiplexer_abci/0.1.0": "bafybeibwshynddwnecxj3prgokmja2hyqgqxvbsllhebzeojee3mmaus2y",
         "contract/valory/mech/0.1.0": "bafybeiauxqogu726oeethvzux6r77zgn7sw5j2l4uigqb47zlqawblkv6e",
         "contract/valory/realitio/0.1.0": "bafybeic5ie4oodetj4krdogydvbfxg4qggc3matpiflocah626tpevpreq",
         "contract/valory/realitio_proxy/0.1.0": "bafybeidx37xzjjmapwacedgzhum6grfzhp5vhouz4zu3pvpgdy5pgb2fr4",

--- a/packages/valory/agents/trader/aea-config.yaml
+++ b/packages/valory/agents/trader/aea-config.yaml
@@ -43,10 +43,10 @@ skills:
 - valory/reset_pause_abci:0.1.0:bafybeibnwjcjx4lluf4cwb6es5peelu3gm7vrzlieygrzpbjlubxpvbvzy
 - valory/termination_abci:0.1.0:bafybeifdtxgldw33kwvsavcituzewwbr6iqfcsgk5qouqfhpwdrivyyyom
 - valory/transaction_settlement_abci:0.1.0:bafybeifpnkwgwpzz6uwrvfgurm26allr6shjfbp7bfbrxwy64sw3nf3fsa
-- valory/tx_settlement_multiplexer_abci:0.1.0:bafybeigvuuvfy7dcdp2kofaorjrqhfxiqdcjsojb4xlcim5x35dmmtzyae
+- valory/tx_settlement_multiplexer_abci:0.1.0:bafybeibwshynddwnecxj3prgokmja2hyqgqxvbsllhebzeojee3mmaus2y
 - valory/market_manager_abci:0.1.0:bafybeifdf6uoscrgvlgqceqcph2pgoygobznw6hbodpav7lubbqsyh4kby
-- valory/decision_maker_abci:0.1.0:bafybeiaan3hccyiocw6a2hecrcfx47sjxkuerbcoxjwjgr32zuqujs3qsu
-- valory/trader_abci:0.1.0:bafybeieb3pleydyefkxd7hcern2ghrlxcyasw2hq76l7lucogtgfjimchy
+- valory/decision_maker_abci:0.1.0:bafybeie6o4n3s557caqxcqtdxquiej6yakwc3z46anbnfnkaguvxkqjkhq
+- valory/trader_abci:0.1.0:bafybeigrm3n43ycgwysx3ovh67drnheyt5c2gqykb7bp7g6eqviyy3utom
 - valory/staking_abci:0.1.0:bafybeigwpp474gw7yb3w3xm4ejz4nxm7n2l4d27po6pj3yh7w3idctge5i
 default_ledger: ethereum
 required_ledgers:

--- a/packages/valory/services/trader/service.yaml
+++ b/packages/valory/services/trader/service.yaml
@@ -7,7 +7,7 @@ license: Apache-2.0
 fingerprint:
   README.md: bafybeigtuothskwyvrhfosps2bu6suauycolj67dpuxqvnicdrdu7yhtvq
 fingerprint_ignore_patterns: []
-agent: valory/trader:0.1.0:bafybeieib2wgae3hgxshw2n6kmii5m76awafe6f46mrkif7wtc5mncvdlm
+agent: valory/trader:0.1.0:bafybeidjit4jcfqr56mcy65jla76g7jiz7nybfoeycftfijtutfqcorfsa
 number_of_agents: 4
 deployment: {}
 ---

--- a/packages/valory/skills/decision_maker_abci/rounds.py
+++ b/packages/valory/skills/decision_maker_abci/rounds.py
@@ -222,6 +222,7 @@ class DecisionMakerAbciApp(AbciApp[Event]):
             get_name(SynchronizedData.utilized_tools),
             get_name(SynchronizedData.redeemed_condition_ids),
             get_name(SynchronizedData.payout_so_far),
+            get_name(SynchronizedData.mech_price),
         }
     )
     final_states: Set[AppState] = {

--- a/packages/valory/skills/decision_maker_abci/skill.yaml
+++ b/packages/valory/skills/decision_maker_abci/skill.yaml
@@ -32,7 +32,7 @@ fingerprint:
   payloads.py: bafybeigcic4vewdglakzpyqevhfwsolh4ywnbvxo5bgned7gl5uo3jif7m
   policy.py: bafybeihca4gc5gdj3wmvtzulqq3cr4zm6ouyt7aoscfedutzr4so4bksna
   redeem_info.py: bafybeifiiix4gihfo4avraxt34sfw35v6dqq45do2drrssei2shbps63mm
-  rounds.py: bafybeigweivru6adchl4vwc6slfuoaknj4ejgo66qm3hmsrxfzyyieocpu
+  rounds.py: bafybeifdqkwosejktea7az6fzzs3gtxkkxcq2kcmgdy4zntgdy42x5sn7q
   states/__init__.py: bafybeid23llnyp6j257dluxmrnztugo5llsrog7kua53hllyktz4dqhqoy
   states/base.py: bafybeiag6q3nlndsu4nt66rxrgogueh364v3lekqo4c5ynuuo4hql2bxum
   states/bet_placement.py: bafybeibalhxhp2c4oljmiwqi6ds3g36fgtabmf42mb5sgq6z22znrcbhda

--- a/packages/valory/skills/trader_abci/skill.yaml
+++ b/packages/valory/skills/trader_abci/skill.yaml
@@ -25,8 +25,8 @@ skills:
 - valory/transaction_settlement_abci:0.1.0:bafybeifpnkwgwpzz6uwrvfgurm26allr6shjfbp7bfbrxwy64sw3nf3fsa
 - valory/termination_abci:0.1.0:bafybeifdtxgldw33kwvsavcituzewwbr6iqfcsgk5qouqfhpwdrivyyyom
 - valory/market_manager_abci:0.1.0:bafybeifdf6uoscrgvlgqceqcph2pgoygobznw6hbodpav7lubbqsyh4kby
-- valory/decision_maker_abci:0.1.0:bafybeiaan3hccyiocw6a2hecrcfx47sjxkuerbcoxjwjgr32zuqujs3qsu
-- valory/tx_settlement_multiplexer_abci:0.1.0:bafybeigvuuvfy7dcdp2kofaorjrqhfxiqdcjsojb4xlcim5x35dmmtzyae
+- valory/decision_maker_abci:0.1.0:bafybeie6o4n3s557caqxcqtdxquiej6yakwc3z46anbnfnkaguvxkqjkhq
+- valory/tx_settlement_multiplexer_abci:0.1.0:bafybeibwshynddwnecxj3prgokmja2hyqgqxvbsllhebzeojee3mmaus2y
 - valory/staking_abci:0.1.0:bafybeigwpp474gw7yb3w3xm4ejz4nxm7n2l4d27po6pj3yh7w3idctge5i
 behaviours:
   main:

--- a/packages/valory/skills/tx_settlement_multiplexer_abci/skill.yaml
+++ b/packages/valory/skills/tx_settlement_multiplexer_abci/skill.yaml
@@ -21,7 +21,7 @@ protocols:
 - valory/ledger_api:1.0.0:bafybeige5agrztgzfevyglf7mb4o7pzfttmq4f6zi765y4g2zvftbyowru
 skills:
 - valory/abstract_round_abci:0.1.0:bafybeicrzndcdbue34yxwwb4hmmdhgzw4owcdcdag3ifj6thpx5wie3dp4
-- valory/decision_maker_abci:0.1.0:bafybeiaan3hccyiocw6a2hecrcfx47sjxkuerbcoxjwjgr32zuqujs3qsu
+- valory/decision_maker_abci:0.1.0:bafybeie6o4n3s557caqxcqtdxquiej6yakwc3z46anbnfnkaguvxkqjkhq
 - valory/staking_abci:0.1.0:bafybeigwpp474gw7yb3w3xm4ejz4nxm7n2l4d27po6pj3yh7w3idctge5i
 behaviours:
   main:

